src/Microsoft.ML.Transforms/TensorflowTransform.cs

```diff
@@ -0,0 +1,665 @@
+            {
+                return CreateTensorValueGetter<string>(input, type, colIndex, tfShape);
+            }
+
+            throw Host.ExceptNotSupp("Tensorflow type not supported");
+        }
+
+        private ITensorValueGetter[] GetTensorValueGetters(IRow input, int iinfo)
+        {
+            var info = _bindings.Infos[iinfo];
+            var tfInfo = _bindings.TfColInfo[iinfo];
+            var srcTensorGetters = new ITensorValueGetter[info.SrcIndices.Length];
+            for (int j = 0; j < info.SrcIndices.Length; j++)
+            {
+                int colIndex = _bindings.Infos[iinfo].SrcIndices[j];
+                srcTensorGetters[j] = CreateTensorValueGetterVec(input, tfInfo.TfTypes[j], info.SrcTypes[j], colIndex, tfInfo.TfShapes[j]);
+            }
+            return srcTensorGetters;
+        }
+
+        private unsafe T[] FetchData<T>(IntPtr data, int size)
+        {
+            var result = new T[size];
+
+            GCHandle handle = GCHandle.Alloc(result, GCHandleType.Pinned);
+            IntPtr target = handle.AddrOfPinnedObject();
+
+            Int64 sizeInBytes = size * Marshal.SizeOf((typeof(T)));
+            Buffer.MemoryCopy(data.ToPointer(), target.ToPointer(), sizeInBytes, sizeInBytes);
+            handle.Free();
+            return result;
+        }
+
+        private Delegate MakeGetter(IRow input, int iinfo)
+        {
+            var info = _bindings.Infos[iinfo];
+            var outInfo = _bindings.OutputCols[iinfo];
+            var tfType = _bindings.OutputTFTypes[iinfo];
+
+            if (tfType == TFDataType.Float)
+            {
+                return Utils.MarshalInvoke(MakeGetter<float>, outInfo.ItemType.RawType, input, iinfo);
+            }
+
+            if (tfType == TFDataType.Double)
+            {
+                return Utils.MarshalInvoke(MakeGetter<double>, outInfo.ItemType.RawType, input, iinfo);
+            }
+
+            if (tfType == TFDataType.Int32)
+            {
+                return Utils.MarshalInvoke(MakeGetter<int>, outInfo.ItemType.RawType, input, iinfo);
+            }
+
+            if (tfType == TFDataType.Int64)
+            {
+                return Utils.MarshalInvoke(MakeGetter<Int64>, outInfo.ItemType.RawType, input, iinfo);
+            }
+
+            if (tfType == TFDataType.Bool)
+            {
+                return Utils.MarshalInvoke(MakeGetter<bool>, outInfo.ItemType.RawType, input, iinfo);
+            }
+
+            if (tfType == TFDataType.String)
+            {
+                return Utils.MarshalInvoke(MakeGetter<string>, outInfo.ItemType.RawType, input, iinfo);
+            }
+
+            throw Host.ExceptNotSupp("Tensorflow type not supported");
+        }
+
+        private ValueGetter<VBuffer<T>> MakeGetter<T>(IRow input, int iinfo)
+        {
+            Host.AssertValue(input);
+            // Need to check type here.
+            //Host.Assert(Infos[iinfo].TypeSrc.IsText);
```

Comment:
>Assert [](start = 19, length = 6)

If you pass in a ColumnType as well here, then you can check that typeof(T)==columnType.ItemType.RawType.